src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -331,7 +329,14 @@ async Task<(Solution Solution, bool IsForked)> GetLspSolutionForWorkspaceAsync(W
 
             // Step 3: Check to see if the LSP text matches the workspace text.
             var documentsInWorkspace = GetDocumentsForUris(_trackedDocuments.Keys.ToImmutableArray(), workspaceCurrentSolution);
-            if (await DoesAllTextMatchWorkspaceSolutionAsync(documentsInWorkspace, cancellationToken).ConfigureAwait(false))
+            var sourceGeneratedDocuments =
+                _trackedDocuments.Keys.Where(static uri => uri.Scheme == SourceGeneratedDocumentUris.Scheme)
+                    .Select(uri => (identity: SourceGeneratedDocumentUris.DeserializeIdentity(workspaceCurrentSolution, uri), _trackedDocuments[uri].Text))
+                    .Where(tuple => tuple.identity.HasValue)
+                    .SelectAsArray(tuple => (tuple.identity!.Value, tuple.Text));
+
+            var doesAllTextMatch = await DoesAllTextMatchWorkspaceSolutionAsync(documentsInWorkspace, cancellationToken).ConfigureAwait(false);
+            if (doesAllTextMatch && !sourceGeneratedDocuments.Any())
```

Comment:
I'm not quite understanding the behavior difference here.  If the source generated text already matches the workspace sln text for source generated documents, why is it not valid to use the workspace sln?

Reply:
Also - could we add a test or two with source generated documents in the lsp workspace manager?  Like making sure we get the expected re-used workspace instance or forked solution instance where appropriate.

Reply:
Ah maybe I need to rename the method here: DoesAllTextMatchWorkspaceSolutionAsync is still checking regular documents only.

Reply:
Kind of feels like DoesAllTextMatchWorkspaceSolutionAsync should check every kind of document and shouldn't special case source gen docs here

Reply:
So this is what's kinda fun here -- it could check if the source generated documents match...by potentially running generators! Otherwise it needs to call the other magic workspace level method that deals with the potential mismatch. This keeps this effectively synchronous in the dispatch queue, since I don't ever want dispatching to be slowing things down.